routes/user.go

```diff
@@ -1811,12 +1811,87 @@ type GetNotificationsRequest struct {
+	PublicKeyBase58Check string
+}
+
+
 type GetNotificationsResponse struct {
 	Notifications       []*TransactionMetadataResponse
 	ProfilesByPublicKey map[string]*ProfileEntryResponse
 	PostsByHash         map[string]*PostEntryResponse
 }
 
+type GetNotificationsCountResponse struct {
+	NotificationsCount uint64
+}
+
+func (fes *APIServer) GetNotificationsCount(ww http.ResponseWriter, req *http.Request) {
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := GetNotificationsCountRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"GetNotificationsCount: Problem parsing request body: %v", err))
+		return
+	}
+
+	userMetadata, err := fes.getUserMetadataFromGlobalState(requestData.PublicKeyBase58Check)
+
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetNotificationsCount: Error getting user metadata from global state: %v", err))
+		return
+	}
+
+	var fetchStartIndex int64
+
+	if userMetadata.LatestUnreadNotificationIndex > 0 {
+		fetchStartIndex = userMetadata.LatestUnreadNotificationIndex + 1
+	} else if userMetadata.NotificationLastSeenIndex > 0 {
+		fetchStartIndex = userMetadata.NotificationLastSeenIndex + 1
+	}
+
+
+	notificationsRequestData := GetNotificationsRequest{
+		PublicKeyBase58Check: requestData.PublicKeyBase58Check,
+		// Only count the notifications that haven't previously been iterated over
+		FetchStartIndex: fetchStartIndex,
+		// If notifications are > 100, we show a "99+" message on the front end.
+		NumToFetch: 100,
+	}
+
+	newNotificationsCount, newNotificationsStartIndex, err := fes._getNotificationsCount(&notificationsRequestData)
+
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetNotificationsCount: Error getting notifications: %v", err))
+		return
+	}
+
+	notificationsCount := newNotificationsCount + userMetadata.UnreadNotifications
+
+	// Update user metadata with new unread notification count.
+	userMetadata.UnreadNotifications = notificationsCount
+	// Save the new latest unread notification index, so that notifications aren't scanned twice. Only do this when new notifications are added.
+	if newNotificationsCount > 0 {
+		userMetadata.LatestUnreadNotificationIndex = newNotificationsStartIndex
+	}
+
+	if err = fes.putUserMetadataInGlobalState(userMetadata); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("AdminResetTutorialStatus: Error putting user metadata in global state: %v", err))
+		return
+	}
+
+	res := &GetNotificationsCountResponse{
+		NotificationsCount: notificationsCount,
+	}
+
+	if err := json.NewEncoder(ww).Encode(res); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"GetNotifications: Problem encoding response as JSON: %v", err))
+		return
+	}
+}
+
+
 func (fes *APIServer) GetNotifications(ww http.ResponseWriter, req *http.Request) {
 	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
 	requestData := GetNotificationsRequest{}
```

Comment:
This function needs to validate that the JWT is signed by the public key that notifications are being requested for (see comment on the GetNotificationsCount function).